### PR TITLE
fix(ci): split insecure-api check from ci; it will be triggered only on PRs

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Fixes master CI. Insecure api job should only run on PRs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -367,24 +367,3 @@ jobs:
           name: ${{ matrix.name }}_report
           path: |
             /tmp/report/
-
```
Comment (FedeDP) [inline .github/workflows/ci.yml:370]: Moved to its own workflow.

File: .github/workflows/insecure-api.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  insecure-api:
+    name: check-insecure-api
+    runs-on: ubuntu-latest
+    container:
+      image: returntocorp/semgrep
+    steps:
+      - name: Checkout Libs ⤵️
+        uses: actions/checkout@v3
```
Comment (FedeDP) [inline .github/workflows/insecure-api.yml:17]: No need to fetch whole history i guess.

Comment (FedeDP) [reply .github/workflows/insecure-api.yml:17]: And i was wrong of course :rofl:

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**